packages/EasyApiPlatform/src/Filter/AdvancedSearchFilter.php

```diff
@@ -0,0 +1,485 @@
+        if ($metadata->hasField($field)) {
+            if ($field === 'id') {
+                $values = array_map([$this, 'getIdFromValue'], $values);
+            }
+
+            if ($this->hasValidValues($values, $this->getDoctrineFieldType($property, $resourceClass)) === false) {
+                $this->logger->notice('Invalid filter ignored', [
+                    'exception' => new InvalidArgumentException(sprintf(
+                        'Values for field "%s" are not valid according to the doctrine type.',
+                        $field
+                    )),
+                ]);
+
+                return;
+            }
+
+            $this->addWhereByStrategy(
+                $strategy,
+                $queryBuilder,
+                $queryNameGenerator,
+                $alias,
+                $field,
+                $values,
+                $caseSensitive
+            );
+
+            return;
+        }
+
+        // metadata doesn't have the field, nor an association on the field
+        if ($metadata->hasAssociation($field) === false) {
+            return;
+        }
+
+        $values = array_map([$this, 'getIdFromValue'], $values);
+
+        $associationResourceClass = (string)$metadata->getAssociationTargetClass($field);
+        $associationFieldIdentifier = $metadata->getIdentifierFieldNames()[0];
+        $doctrineTypeField = $this->getDoctrineFieldType($associationFieldIdentifier, $associationResourceClass);
+
+        if ($this->hasValidValues($values, $doctrineTypeField) === false) {
+            $this->logger->notice('Invalid filter ignored', [
+                'exception' => new InvalidArgumentException(sprintf(
+                    'Values for field "%s" are not valid according to the doctrine type.',
+                    $field
+                )),
+            ]);
+
+            return;
+        }
+
+        $associationAlias = $alias;
+        $associationField = $field;
+        if (
+            $metadata->isCollectionValuedAssociation($associationField)
+            || $metadata->isAssociationInverseSide($field)
+        ) {
+            $associationAlias = QueryBuilderHelper::addJoinOnce(
+                $queryBuilder,
+                $queryNameGenerator,
+                $alias,
+                $associationField
+            );
+            $associationField = $associationFieldIdentifier;
+        }
+
+        $this->addWhereByStrategy(
+            $strategy,
+            $queryBuilder,
+            $queryNameGenerator,
+            $associationAlias,
+            $associationField,
+            $values,
+            $caseSensitive
+        );
+    }
+
+    /**
+     * Gets the ID from an IRI or a raw ID.
+     */
+    protected function getIdFromValue(string $value): int|string
+    {
+        try {
+            $iriConverter = $this->getIriConverter();
+            $item = $iriConverter->getResourceFromIri($value, ['fetch_data' => false]);
+
+            return $this->getPropertyAccessor()
```

Comment:
Can we write it inline?

Reply:
Method removed in favor of `\ApiPlatform\Doctrine\Common\Filter\SearchFilterTrait`